src/main/java/de/rabea/game/Board.java

```diff
@@ -101,7 +118,7 @@ public boolean isIndexOfLastCell(int cell) {
         return cell == indexOfLastCell();
     }
 
-    public boolean isIndexOfEndOfFirstOrSecondRow(int cell) {
+    public boolean isEndOfRowIndexExceptLastRow(int cell) {
```

Comment:
This method name feels overly explicit. I feel like there should be a more 'comfortable' way that doesn't need such special-casing...